### PR TITLE
Adding containerd k8s http_server example

### PR DESCRIPTION
**Pull request by AvengerMoJo**
Adding the crun patch and now it could run with contained on top of the k8s. 

Signed-off-by: Alex Lau (AvengerMoJo) <avengermojo@gmail.com>

### CODE REVIEW

File: containerd/http_server/full-cluster/README.md [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+
+Containerd uses the `runc` runtime by default as well and we need to configure it to use `crun` instead.
+That is done by adding to two configuration files.
+
+First, create a `/etc/containerd/config.toml` file by running
+
+```bash
+
+sudo mkdir -p /etc/containerd/
+sudo containerd config default > /etc/containerd/config.toml
```
Comment (0xE282B0) [inline containerd/http_server/full-cluster/README.md]: I get `-bash: /etc/containerd/config.toml: Permission denied` when running as user.
```suggestion
sudo bash -c "containerd config default > /etc/containerd/config.toml"
```

File: containerd/http_server/full-cluster/README.md [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+  [plugins."io.containerd.runtime.v1.linux"]
+    no_shim = false
+    runtime = "crun"
+
+```
+
+Next, restart Containerd to apply the configuration changes.
+
+```bash
+systemctl restart containerd
```
Comment (0xE282B0) [inline containerd/http_server/full-cluster/README.md]: Just for convenience.
```suggestion
sudo systemctl restart containerd
```

File: containerd/http_server/full-cluster/README.md [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+We can now run the WebAssembly-based image from Docker Hub in the Kubernetes cluster.
+By applying the [yaml script](../../../kubernetes_crio/http_server/full-cluster/k8s-http_server.yaml) and run it.
+
+```bash
+kubectl apply -f k8s-http_server.yaml
+k8s-http_server.yaml
+
+mubectl get pod --all-namespaces -o wide
+NAMESPACE     NAME                            READY   STATUS    RESTARTS   AGE    IP           NODE    NOMINATED NODE   READINESS GATES
+default       http-server                     1/1     Running   0          3s     <$host_ip>   k8s-1   <none>           <none>
```
Comment (0xE282B0) [inline containerd/http_server/full-cluster/README.md]: When I apply the manifest I get a CrashLoopBackOff.
```
NAMESPACE    NAME           READY   STATUS             RESTARTS   AGE    IP           NODE                 NOMINATED NODE   READINESS GATES
default      http-server    0/1     CrashLoopBackOff   4          115s   172.18.0.3   kind-control-plane   <none>           <none>
```
The Container has the following log output:
```
[2021-12-15 10:28:42.597] [error] instantiation failed: incompatible import type, Code: 0x61                                                                                                            
[2021-12-15 10:28:42.597] [error]     Mismatched function type. Expected: FuncType {params{i32 , i32 , i32} returns{i32}} , Got: FuncType {params{i32 , i32} returns{i32}}                              
[2021-12-15 10:28:42.597] [error]     When linking module: "wasi_snapshot_preview1" , function name: "sock_accept"                                                                                      
[2021-12-15 10:28:42.597] [error]     At AST node: import description                                                                                                                                   
[2021-12-15 10:28:42.597] [error]     At AST node: import section                                                                                                                                       
[2021-12-15 10:28:42.597] [error]     At AST node: module                                                                                                                                               
runc: could not get wasmedge result from VM  
```
Am I doing something wrong?

File: containerd/http_server/full-cluster/README.md [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+In this section, we will start from **another terminal window** and start using the cluster.
+
+We can now run the WebAssembly-based image from Docker Hub in the Kubernetes cluster.
+By applying the [yaml script](../../../kubernetes_crio/http_server/full-cluster/k8s-http_server.yaml) and run it.
+
+```bash
+kubectl apply -f k8s-http_server.yaml
+k8s-http_server.yaml
+
+mubectl get pod --all-namespaces -o wide
```
Comment (0xE282B0) [inline containerd/http_server/full-cluster/README.md]: Typo:
```suggestion
kubectl get pod --all-namespaces -o wide
```

**Review by 0xE282B0 — COMMENTED** [review]
Hi, it's awesome to see how this repo evolves!

I was able to reproduce these steps except that I get an error when I try to start the http_server, but I get the same error when using the CRI-O based Kubernetes cluster.
Could that be a problem with the Image on DockerHub?

Great work. I love to see that there is a way to use wasmedge with containerd and K8s.

File: containerd/http_server/full-cluster/README.md (hunk 3 above)
Comment (hydai) [reply containerd/http_server/full-cluster/README.md]: @AvengerMoJo 

Please update to the latest WasmEdge release 0.9.0 and use the latest crun master. Also, the example should compile with the latest wasmedge_wasi_socket release.

The root cause is that the original sock_accept implementation has some problem which is fixed after 0.9.0. And this also changes the function signature.

Comment (0xE282B0) [reply containerd/http_server/full-cluster/README.md]: Thanks for the quick response!

I created an Image with [this guide](https://github.com/second-state/wasmedge-containers-examples/blob/main/http_server_wasi_app.md) and everything is working!

Only the imgae from the example (`avengermojo/http_server:with-wasm-annotation`) seems to be broken.

Comment (hydai) [reply containerd/http_server/full-cluster/README.md]: Yes. We just released the new version of WasmEdge last Friday, which is later than this PR. So the example will need to update again :-)

**Pull request merged**